src/components/Form/Input.tsx

```diff
@@ -0,0 +1,47 @@
+import { forwardRef, type DetailedHTMLProps, type InputHTMLAttributes, type ReactNode } from "react";
+import { twJoin } from "tailwind-merge";
+import "./Input.css";
+
+export interface InputProps
+  extends Omit<DetailedHTMLProps<InputHTMLAttributes<HTMLInputElement>, HTMLInputElement>, "prefix" | "suffix"> {
+  className?: string;
+  wrapperClassName?: string;
+  decorators?: {
+    prefix?: ReactNode;
+    suffix?: ReactNode;
+  };
+  disabled?: boolean;
+  state?: {
+    type?: "error" | "warning";
+    text?: string;
+    render?: ReactNode;
+  };
+}
+
+export const Input = forwardRef<HTMLInputElement, InputProps>(
+  ({ className, wrapperClassName, decorators, disabled = false, state, ...props }, ref) => {
+    const stateClass =
+      state?.type === "error" ? "bbn-input-error" : state?.type === "warning" ? "bbn-input-warning" : "";
+
+    const stateTextClass =
+      state?.type === "error"
+        ? "bbn-input-state-text-error"
+        : state?.type === "warning"
+          ? "bbn-input-state-text-warning"
+          : "";
```

Comment:
Try to avoid of using `if/else` or `switch/case` statements when ever it's possible. Use collections instead (GoF: strategy pattern)

Reply:
```suggestion
```